packages/plugin-core/src/commands/MoveNoteCommand.ts

```diff
@@ -316,10 +322,97 @@ export class MoveNoteCommand extends BasicCommand<CommandOpts, CommandOutput> {
     panel.webview.html = md.render(contentLines.join("\n"));
   }
 
-  addAnalyticsPayload(_opts?: CommandOpts, out?: CommandOutput) {
-    const payload =
+  async trackProxyMetrics({
+    opts,
+    noteChangeEntryCounts,
+  }: {
+    opts: CommandOpts;
+    noteChangeEntryCounts: {
+      createdCount?: number;
+      deletedCount?: number;
+      updatedCount?: number;
+    };
+  }) {
+    const extension = ExtensionProvider.getExtension();
+    const engine = extension.getEngine();
+    const { vaults } = engine;
+
+    // since we allow multiples, aggregate them.
+    const { moves } = opts;
+
+    let numProcessedAcc = 0;
+    const numChildrenAcc: number[] = [];
+    const numLinksAcc: number[] = [];
+    const numCharsAcc: number[] = [];
+    const noteDepthAcc: number[] = [];
+    let traitsAcc: string[] = [];
+    await asyncLoop(moves, async (move) => {
+      const { fname, vaultName: vname } = move.newLoc;
+      if (fname !== undefined && vname !== undefined) {
+        const newLocVault = VaultUtils.getVaultByName({ vaults, vname });
+        const newLocNote = (
+          await engine.findNotes({
```

Comment:
Because the outputted `NoteChangeEntry[]` doesn't hold enough context to track back the actual transactions made (or is non-trivial and unreliable to try to do so). 

There are cases for move note (and all other commands you suggested to use `NoteChangeEntry` to get notes) that will output multiple `create` entries for a single moved note. 

for example, cases where move note results in a creation of a stub in place of the old note.

I could filter for that case specifically, but rather than try to find all edge cases and filtering them out of the list of note change entries, relying on the actual moves passed to `.execute` as a command option would be the surest way to make sure we are only grabbing metrics from the right place.

@jonathanyeung and I briefly discussed the state of the events being emitted in our engine, and that just reading them doesn't give us the most accurate / high level picture of what has happened. We haven't come to any solution but I think some enhancements in this area is needed to just rely on note change entries for now.

As an interim solution, I will instead grab these information from during the execution, and pass it around so that it eventually ends up in the commands' respective `CommandOutput`s

Reply:
in this case, I would rather we change the output to include both `NoteChangeEntry` as well as the original input. this is also the pattern for all AWS calls. this way, instead of calling a `findNotes` with the engine, we can grab it from the result.

long term, for `NoteChangeEntry`, does it make sense to separate entries by `sideEffect` vs `command`?

Reply:
that would make sense, and also would be preferable if it is enhanced in a way (either by the way it is, or by how we emit the change entries everywhere) that piecing all change entries together we can derive the pre-operation state. Either way, this needs a proper separate discussion.

Reply:
hold up - so whats the resolution here? until we kick of an actual async or address it in code, can we leave this open?

Reply:
@kevinslin 
I don't think this issue should be a blocking one.

Just for clarification, after I made changes (https://github.com/dendronhq/dendron/pull/3104#issuecomment-1162012062) I removed all `engine.findNotes` calls except for one case where `moveNote` is not a multi move. It is a single call to find one note.

In every other cases I changed it so that we derive the payload from what we can infer from the initial input and intermediate steps.